lldb/source/API/SBTarget.cpp

```diff
@@ -2039,7 +2039,17 @@ lldb::SBInstructionList SBTarget::ReadInstructions(lldb::SBAddress base_addr,
       const size_t bytes_read =
           target_sp->ReadMemory(*addr_ptr, data.GetBytes(), data.GetByteSize(),
                                 error, force_live_memory, &load_addr);
+
       const bool data_from_file = load_addr == LLDB_INVALID_ADDRESS;
+      if (!flavor_string || flavor_string[0] == '\0') {
```

Comment:
This check already exists in `Disassembler::FindPluginForTarget`, maybe more straightforward would be to call a function in the `Disassembler` class that uses it instead of `Disassembler::DisassembleBytes`?